### PR TITLE
refactor write operation, use reference in delete operations

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:
#202 
* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: engine/src/engine_api.rs [hunk 1]
```diff
@@ -1,13 +1,87 @@
 use crate::error::EngineError;
 
+// /// Write operation
```
Comment (rogercloud) [inline engine/src/engine_api.rs:3]: Remove the useless comments.

File: engine/src/engine_api.rs [hunk 2]
```diff
@@ -1,13 +1,87 @@
...
 #[derive(Debug)]
-pub enum WriteOperation {
+pub enum WriteOperation<'a> {
     /// `Put` operation
     Put {
         /// The table name
-        table: &'static str,
+        table: &'a str,
         /// Key
         key: Vec<u8>,
```
Comment (rogercloud) [inline engine/src/engine_api.rs:86]: What's special for the `Put` operation for that the vector type is kept?

**Review by rogercloud — COMMENTED** [review]
Actually this PR also removes the buffer_ops, which is a big part in the PR, please comment this change in the commit message.

File: engine/src/engine_api.rs (hunk 1 above)
Comment (themanforfree) [reply engine/src/engine_api.rs]: Fixed

**Review by rogercloud — APPROVED** [review]

**Pull request merged**